src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -28,8 +30,14 @@ public DeleteCommand<?> parse(String args) throws ParseException {
 
 
         String entity = ParserUtil.parseEntity(entityString);
-        Index index = ParserUtil.parseIndex(indexString);
 
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(indexString);
+        } catch (ParseException e) {
+            throw new ParseException(MESSAGE_INVALID_INDEX);
+        }
```

Comment:
Do not use this exception.  

parseIndex already validate the index and will throw exception if necessary.

Reply:
Roger, will remove this check